app/models/concerns/operations/stock_operations.rb

```diff
@@ -112,14 +114,15 @@ def register_loss(package, quantity:, location: nil, action: 'loss', description
     # @param [String] action: the inventory action (optional)
     # @param [String] description: notes to detail the action
     #
-    def register_quantity_change(package, quantity:, location:, action:, description: nil)
+    def register_quantity_change(package, quantity:, location:, action:, description: nil, source: nil)
```

Comment:
we can make the params list to a hash

Reply:
If you mean doing `def register_quantity_change(package, opts)` then I'd rather not. Because that would make all of them optional. Ruby makes those that don't have a default value are mandatory